File: modules/day-03-reproducible/r/solution.R
measurements <- data.frame(
  sample_id = c("S1", "S2", "S3", "S4"),
  condition = c("control", "control", "treated", "treated"),
  signal = c(5.2, 5.5, 6.8, 7.1)
)

summarize_condition <- function(data, condition_name) {
  mean(data$signal[data$condition == condition_name])
}

run_analysis <- function(input_data) {
  treated_mean <- summarize_condition(input_data, "treated")
  control_mean <- summarize_condition(input_data, "control")

  list(
    treated_mean = treated_mean,
    control_mean = control_mean,
    mean_difference = treated_mean - control_mean
  )
}

analysis_result <- run_analysis(measurements)

cat("Required packages: base R only\n")
cat("Inputs: a data frame with sample_id, condition, and signal columns\n")
cat("Output: difference in mean signal between treated and control samples\n")
cat("Mean difference:", analysis_result$mean_difference, "\n")

File: modules/day-03-reproducible/python/solution.py
measurements = [
    {"sample_id": "S1", "condition": "control", "signal": 5.2},
    {"sample_id": "S2", "condition": "control", "signal": 5.5},
    {"sample_id": "S3", "condition": "treated", "signal": 6.8},
    {"sample_id": "S4", "condition": "treated", "signal": 7.1},
]


def mean_signal(rows, condition_name):
    values = [row["signal"] for row in rows if row["condition"] == condition_name]
    return sum(values) / len(values)


def run_analysis(rows):
    treated_mean = mean_signal(rows, "treated")
    control_mean = mean_signal(rows, "control")
    return {
        "treated_mean": treated_mean,
        "control_mean": control_mean,
        "mean_difference": treated_mean - control_mean,
    }


if __name__ == "__main__":
    result = run_analysis(measurements)
    print("Required packages: standard library only")
    print("Inputs: a list of records with sample_id, condition, and signal")
    print("Output: difference in mean signal between treated and control")
    print("Mean difference:", result["mean_difference"])
